# 01 — Data Understanding
HealthConnect Appointment No-Show Prediction — Week 5

In [1]:
import sys
sys.path.insert(0, '..')

from src.data.load_data import load_appointment_data
from src.data.validate_data import validate_dataset
from src import config

df = load_appointment_data(config.RAW_DATA_PATH)
df.head()

,appointment_id,patient_id,gender,age,age_group,appointment_type,booking_date,appointment_date,appointment_day,appointment_time,booking_lead_days,previous_appointments,previous_no_shows,reminder_sent,reminder_channel,distance_to_clinic_km,waiting_time_minutes,appointment_outcome
0,HC-00001,P-1613,Female,39,35-44,Follow-up,2/6/2025,2/18/2025,Tuesday,Afternoon,12,2,0,Yes,WhatsApp,19.3,29.0,No-Show
1,HC-00002,P-0813,Male,31,25-34,Specialist Consultation,2/25/2026,2/27/2026,Friday,Morning,2,6,0,Yes,SMS,14.3,42.0,Attended
2,HC-00003,P-1366,Female,50,45-54,General Consultation,11/16/2025,12/24/2025,Wednesday,Morning,38,5,1,Yes,SMS,11.4,11.0,No-Show
3,HC-00004,P-1031,Male,59,55-64,Follow-up,7/18/2025,8/28/2025,Thursday,Evening,41,3,1,Yes,SMS,7.4,35.0,Attended
4,HC-00005,P-1458,Female,34,25-34,Follow-up,7/9/2025,8/25/2025,Monday,Afternoon,47,3,1,Yes,Email,5.6,27.0,No-Show


In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   appointment_id         5000 non-null   str    
 1   patient_id             5000 non-null   str    
 2   gender                 5000 non-null   str    
 3   age                    5000 non-null   int64  
 4   age_group              5000 non-null   str    
 5   appointment_type       5000 non-null   str    
 6   booking_date           5000 non-null   str    
 7   appointment_date       5000 non-null   str    
 8   appointment_day        5000 non-null   str    
 9   appointment_time       5000 non-null   str    
 10  booking_lead_days      5000 non-null   int64  
 11  previous_appointments  5000 non-null   int64  
 12  previous_no_shows      5000 non-null   int64  
 13  reminder_sent          5000 non-null   str    
 14  reminder_channel       3634 non-null   str    
 15  distance_to_cli

In [3]:
df.describe(include='all').T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
appointment_id,5000,5000,HC-00001,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN
patient_id,5000,1696,P-1776,9,NaN,NaN,NaN,NaN,NaN,NaN,NaN
gender,5000,3,Female,2488,NaN,NaN,NaN,NaN,NaN,NaN,NaN
age,5000.0,NaN,NaN,NaN,48.7948,18.138547,18.0,33.0,49.0,64.0,80.0
age_group,5000,6,65+,1241,NaN,NaN,NaN,NaN,NaN,NaN,NaN
appointment_type,5000,4,General Consultation,2086,NaN,NaN,NaN,NaN,NaN,NaN,NaN
booking_date,5000,593,8/24/2025,20,NaN,NaN,NaN,NaN,NaN,NaN,NaN
appointment_date,5000,545,3/14/2026,18,NaN,NaN,NaN,NaN,NaN,NaN,NaN
appointment_day,5000,7,Wednesday,737,NaN,NaN,NaN,NaN,NaN,NaN,NaN
appointment_time,5000,3,Morning,2227,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [4]:
report = validate_dataset(df)
print(report.summary())

HealthConnect Data Validation Report
Rows: 5000 | Columns: 18
Duplicate appointment_id count: 0
Fully duplicate rows: 0
Invalid appointment_outcome values: 0
Invalid reminder_sent values: 0
Invalid appointment_time values: 0
Unparseable dates: 0
previous_no_shows > previous_appointments: 0

Missing value counts:
  - reminder_channel: 1366
  - distance_to_clinic_km: 90
  - waiting_time_minutes: 60

Warnings:
  - booking_date/appointment_date parse successfully but are stored as M/D/YYYY text, not the ISO format stated in the data dictionary. Downstream code should not assume ISO format.

Overall result: PASSED


## Observations
- 5,000 rows, 18 columns, matches the data dictionary.
- Missing values: `reminder_channel` (1,366), `distance_to_clinic_km` (90), `waiting_time_minutes` (60) — all expected per the Week 4 design doc.
- `booking_date`/`appointment_date` are `M/D/YYYY` text, not ISO as the dictionary states.
- Target distribution (before excluding Cancelled): see next cell.

In [5]:
df['appointment_outcome'].value_counts()

appointment_outcome
No-Show      2423
Attended     2314
Cancelled     263
Name: count, dtype: int64